### ISSUE TITLE
Feature: `sort-object-types` add support for `newlines-between`

### ISSUE THREAD

**Issue by olegafx** [azat-io/eslint-plugin-perfectionist#146]
### What rule do you want to change?

sort-object-types

### Describe the problem

There is no support for `newlines-between`, so it's not possible to automatically add newlines between custom groups.

### Code example

For example, we have a type like this:
```
type Person = {
  onBirthday: () => void;
  name: string;
  onAgeChange: (age: number) => void;
  age?: number;
  surname: string;
  birthday?: Date;
}
```

I expect it to be sorted like this using custom groups:
```
type Person = {
  name: string;
  surname: string;
  
  age?: number;
  birthday?: Date;
  
  onAgeChange: (age: number) => void;
  onBirthday: () => void;
}
```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Added in 69b643e.
Released in [v3.7.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.7.0).

**Comment by olegafx** [comment]
Thanks! However, I still can't figure out how to set up the plugin for my original example.

Even this type is marked as correct:
```
export type Person = {
  name: string;
  age?: number;
};
```
with this configuration:
```
{
  type: 'natural',
  newlinesBetween: 'always',
  groupKind: 'required-first',
  groups: [['unknown', 'multiline'], 'method'],
}
```
But if I add a method, it will divide into groups:
```
export type Person = {
  name: string;

  onAgeChange: (age: number) => void;

  age?: number;
};
```

I was trying to set up `customGroups` for ['requiredType', 'optionalType', 'requiredMethod', 'optionalMethod'], but it didn't work either.

**Comment by azat-io** [comment]
@olegafx That's right, that's how it should work. Can you show your full ESLint config?

**Comment by olegafx** [comment]
I was using the configuration from the previous message and some versions with custom groups, but I didn't save them.

But how to make it to turn this:
```
type Person = {
  onBirthday: () => void;
  name: string;
  onAgeChange: (age: number) => void;
  age?: number;
  surname: string;
  onBirthdayChange?: (birthday: Date) => void;
  birthday?: Date;
}
```
Into this:
```
type Person = {
  name: string;
  surname: string;
  
  age?: number;
  birthday?: Date;
  
  onAgeChange: (age: number) => void;
  onBirthday: () => void;

  onBirthdayChange?: (birthday: Date) => void;
}
```
?

**Comment by hugop95** [comment]
This is related for me to #395 and #217: the drawback of having a `groupKind` option distinct from `groups` is that you have to choose between:
- Sorting by `groups` then `groupKind`.
- Sorting by `groupKind` then `groups`.

Currently, we sort by `groups` then `groupKind`: you will either have all optional attributes above required attributes, or the opposite.

What you are trying to do is impossible with `groupKind`.

> I was trying to set up customGroups for ['requiredType', 'optionalType', 'requiredMethod', 'optionalMethod'], but it didn't work either.

There is a proposal to handle something like this in #395 (https://github.com/azat-io/eslint-plugin-perfectionist/issues/395#issuecomment-2496248187) with a system similar to `sort-classes`, but this is not yet implemented.

As of today, you can use `newlinesBetween` to separate groups yourself with a newline, but that's obviously not ideal.

**Comment by azat-io** [comment]
Implemented in 8407bae.
Released in [v4.2.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.2.0).
